src/layout.rs

```diff
@@ -1133,6 +1138,35 @@ mod tests {
                     Rect::new(0, 1, 1, 0)
                 ]
             );
+
+            // minimal bug from
+            // https://github.com/ratatui-org/ratatui/pull/404#issuecomment-1681850644
```

Comment:
maybe mention what this test is meant to test (that start is below end and below area length?)

Reply:
I will definitely add a few more tests that hit more edge cases more explicitly in another PR. I'm going to merge this now to fix the bug and come back to this.